### PR TITLE
GS-84: Added tests and some improvements.

### PR DESCRIPTION
**Pull request by ydrozhdzhal**

### CODE REVIEW

File: receiver/googlecloudspannerreceiver/metricsreader.go [hunk 1]
```diff
@@ -25,6 +25,12 @@ import (
 	"google.golang.org/api/iterator"
 )
 
+const (
+	projectIdLabelName = "project_id"
```
Comment (kostia-hospodarysko) [inline receiver/googlecloudspannerreceiver/metricsreader.go]: Does it make sense to add some prefix to labels, like "gcp", "sp" or "gcp_sp" or msth like that to denote that labels are domain specific?

**Review by kostia-hospodarysko — APPROVED** [review]

File: receiver/googlecloudspannerreceiver/metricsreader.go (hunk 1 above)
Comment (ydrozhdzhal) [reply receiver/googlecloudspannerreceiver/metricsreader.go]: We agreed that label names will be exactly the same as in Cloud Monitoring, and currently they have the same names as in CM.

File: receiver/googlecloudspannerreceiver/metricsreader.go [hunk 2]
```diff
@@ -247,26 +253,18 @@ func (metadata *MetricsReaderMetadata) ReadMetrics(ctx context.Context, client *
 		if err == iterator.Done {
```
Comment (asukhyy) [inline receiver/googlecloudspannerreceiver/metricsreader.go]:
```
if err != nil {
  if err == iterator.Done {
   break
  }
  logger.Error(fmt.Sprintf("Query %v failed with %v", metadata.Query, err))
  return nil, err
}
```

File: receiver/googlecloudspannerreceiver/metricsreader.go [hunk 3]
```diff
@@ -344,10 +342,39 @@ func toMetricValue(metadata MetricValueMetadata, row *spanner.Row) (MetricValue,
 	return value, err
 }
 
-func (metadata *MetricsReaderMetadata) toMetrics(intervalEnd time.Time, labelValues []LabelValue, metricValues []MetricValue) []pdata.Metrics {
-	metrics := make([]pdata.Metrics, len(metricValues))
+func (metadata *MetricsReaderMetadata) rowToMetrics(logger *zap.Logger, row *spanner.Row) ([]pdata.Metrics, error) {
```
Comment (asukhyy) [inline receiver/googlecloudspannerreceiver/metricsreader.go]: logger is not needed here - just provide error w/ right message.

**Review by asukhyy — APPROVED** [review]

**Pull request merged**